QuantConnect.IBAutomater/IBAutomater.cs

```diff
@@ -207,6 +250,32 @@ public void Stop()
                     // ignored
                 }
             }
+
+            _processId = 0;
+        }
+
+        /// <summary>
+        /// Returns whether the IBGateway is running
+        /// </summary>
+        /// <returns>true if the IBGateway is running</returns>
+        public bool IsRunning()
+        {
+            if (_processId == 0)
+            {
+                return false;
+            }
+
+            try
+            {
+                Process.GetProcessById(_processId);
```

Comment:
Could also keep the `Process` instance and check `Process.HasExited`?
just checking: the `IBAutomater.bat/IBAutomater.sh` **don't** start another process and return right away, but instead, wait until it finishes?

Reply:
Correct, the bat/sh process does not terminate until the child process is terminated.